lifecycle/src/policy.rs

```diff
@@ -42,6 +42,111 @@ where
         }
     }
 
+    /// Check if database exceeds memory limits and free memory if necessary
+    fn check_memory(&mut self, soft_limit: usize, drop_non_persisted: bool) {
```

Comment:
```suggestion
    fn check_memory_and_drop_chunks(&mut self, soft_limit: usize, drop_non_persisted: bool) {
```

Or maybe
```suggestion
    fn enforce_memory_limit(&mut self, soft_limit: usize, drop_non_persisted: bool) {
```

I am just thinking `check_memory` makes it sound like the code is just checking the limits, but in reality it is also attempting to free up memory as well

Reply:
I went with `maybe_free_memory`